kyo-kernel/shared/src/main/scala/kyo/kernel/Loop.scala

```diff
@@ -602,14 +600,18 @@ object Loop:
       * @return
       *   Nothing, as this loop runs forever unless interrupted
       */
-    inline def forever[S](inline run: Safepoint ?=> Unit < S)(using Frame, Safepoint): Unit < S =
-        def _loop(): Unit < S = loop()
-        @tailrec def loop(): Unit < S =
-            run match
-                case kyo: Kyo[Unit, S] @unchecked =>
-                    kyo.andThen(_loop())
```

Comment:
Avoid `andThen` since it has more overhead than instantiating `KyoSuspend` directly, which is also the pattern in the other methods